cpplint.py

```diff
@@ -50,12 +50,23 @@
 import sre_compile
 import string
 import sys
+import itertools
 import unicodedata
 
+# Files with any of these extensions are considered to be
+# header files (and will undergo different style checks).
+# This set can be extended by using the --headers
+# option (also supported in CPPLINT.cfg)
+_header_extensions = set(['h', 'hpp', 'hxx', 'h++', 'cuh'])
+_nonheader_extensions = set(['c', 'cc', 'cpp', 'cxx', 'c++', 'cu'])
+
+
 # The allowed extensions for file names
 # This is set by --extensions flag.
-_valid_extensions = set(['c', 'cc', 'cpp', 'cxx', 'c++', 'h', 'hpp', 'hxx',
-    'h++'])
+_valid_extensions = _nonheader_extensions.union(_header_extensions)
```

Comment:
fair enough. I want to stay backwards compatible, to changing the --extension option is not that great. I added a commit that circumvents the problem by not storing in global variables a mixture of defaults and user-specified files. Instead, at runtime functions evaluate those sets. This is more painful to review I guess, but you asked for it. Also this reminds me that I would like unit tests for all these file filters, but I won't have the time soon.